include/deal.II/lac/shifted_matrix.h

```diff
@@ -213,51 +213,51 @@ ShiftedMatrix<MatrixType>::residual (VectorType       &dst,
 
 
 //---------------------------------------------------------------------------
-template <typename MatrixType, class MASSMatrixType, class VectorType>
+template <typename MatrixType, class MatrixType2, class VectorType>
 inline
-ShiftedMatrixGeneralized<MatrixType, MASSMatrixType, VectorType>
-::ShiftedMatrixGeneralized (const MatrixType     &A,
-                            const MASSMatrixType &M,
-                            const double          sigma)
+ShiftedMatrixGeneralized<MatrixType, MatrixType2, VectorType>
+::ShiftedMatrixGeneralized (const MatrixType  &A,
+                            const MatrixType2 &M,
+                            const double       sigma)
   :
   A(&A), M(&M), sigma(sigma)
 {}
 
 
-template <typename MatrixType, class MASSMatrixType, class VectorType>
+template <typename MatrixType, class MatrixType2, class VectorType>
 inline void
-ShiftedMatrixGeneralized<MatrixType, MASSMatrixType, VectorType>::shift (const double s)
+ShiftedMatrixGeneralized<MatrixType, MatrixType2, VectorType>::shift (const double s)
 {
   sigma = s;
 }
 
-template <typename MatrixType, class MASSMatrixType, class VectorType>
+template <typename MatrixType, class MatrixType2, class VectorType>
 inline double
-ShiftedMatrixGeneralized<MatrixType, MASSMatrixType, VectorType>::shift () const
+ShiftedMatrixGeneralized<MatrixType, MatrixType2, VectorType>::shift () const
 {
   return sigma;
 }
 
 
-template <typename MatrixType, class MASSMatrixType, class VectorType>
+template <typename MatrixType, class MatrixType2, class VectorType>
 inline void
-ShiftedMatrixGeneralized<MatrixType, MASSMatrixType, VectorType>::vmult
+ShiftedMatrixGeneralized<MatrixType, MatrixType2, VectorType>::vmult
 (VectorType       &dst,
  const VectorType &src) const
 {
   A->vmult(dst, src);
   if (sigma != 0.)
     {
-      aux.reinit(dst);
-      M.vmult(aux, src);
+      aux.reinit(dst.size());
```

Comment:
A reason why you change the initialization this way? This will limit the class do serial vectors that can be initialized by size. The generic way is to call reinit on an exemplar (dst in this case).